tests/communication_1to2_test.py

```diff
@@ -0,0 +1,95 @@
+import os
```

Comment:
Why did you split them into several files? We can have a single communication script file. We can trigger the different test cases using flags.